packages/cli/src/services/deployService.js

```diff
@@ -44,19 +46,45 @@ const upload = (url, data, apiKey, onUploadProgress) => {
         try {
           const parsedData = JSON.parse(rawData);
 
-          if (res.statusCode >= 200 && res.statusCode <= 300) {
+          if (res.statusCode >= 200 && res.statusCode < 300) {
             resolve(parsedData);
+          } else if (res.statusCode >= 300 && res.statusCode < 400) {
+            reject(`The server attempted an unsupported redirect. [${res.statusCode}]`);
+          } else if (res.statusCode >= 400 && res.statusCode < 500) {
+            switch (res.statusCode) {
+              case 400:
+                reject("Error: Failed to deploy. [400]");
+                break;
+              case 401:
+                reject(`Error: The API key was not accepted by ${url} [401]`);
+              default:
+                reject(`Error: Failed to deploy with an unknown error. [${res.statusCode}]`);
+            }
+          } else if (res.statusCode >= 500 && res.statusCode < 600) {
+            switch (res.statusCode) {
+              case 500:
+                reject(`Error: The SPAship has encountered a mysterious problem; someone call Richard Feynman! [500]`);
```

Comment:
Did you mean: 
```suggestion
                reject(`Error: The SPAship CLI has encountered a mysterious problem; someone call Richard Feynman! [500]`);
```

Reply:
Actually I meant "The SPAship server" but good catch nonetheless!

```suggestion
                reject(`Error: The SPAship server has encountered a mysterious problem; someone call Richard Feynman! [500]`);
```